### PR TITLE
[cisco_duo] Fix bad value in ip and date fields

### PR DESCRIPTION
**Pull request by vinit-elastic**
Type of change
- Enhancement

## What does this PR do?

This PR adds support for handling bad values in ip and date fields.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's changelog.yml file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Closes https://github.com/elastic/integrations/issues/3696

## Screenshots

image
image
image
image

### CODE REVIEW

File: packages/cisco_duo/data_stream/admin/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -24,9 +24,13 @@ processors:
   - date:
       field: json.timestamp
       target_field: "@timestamp"
-      ignore_failure: true
+      if: ctx.json?.timestamp != null
       formats:
         - UNIX
+      on_failure:
+        - append:
```
[the remark this answers is not in the document]
Comment (vinit-elastic) [reply packages/cisco_duo/data_stream/admin/elasticsearch/ingest_pipeline/default.yml:31]: Oh yeah, I missed it. I'll update it.

**Review by andrewkroh — APPROVED** [review]

**Comment by andrewkroh** [comment]
@vinit-elastic Is there test case with an IP field containing "None" already in the package?

**Comment by vinit-elastic** [comment]
> @vinit-elastic Is there test case with an IP field containing "None" already in the package?

Hey @andrewkroh, 
We didn't add the log because adding log in pipeline test would make the test fail with the following error. 
`unexpected pipeline error (unexpected error.message type []interface {}): ['None' is not an IP string literal.]`

**Comment by andrewkroh** [comment]
Ok, thanks. So in order to support this type of testing we need a way to ignore the error.message from the test config.

**Comment by andrewkroh** [comment]
Related: https://github.com/elastic/elastic-package/issues/912

**Pull request merged**